src/Core/Entities/PlayerEntity.m

```diff
@@ -7911,21 +7911,24 @@ - (NSString *) primedEquipmentName:(NSInteger)offset
 }
 
 
-- (BOOL) setPrimedEquipment:(NSString *)eqKey
+- (BOOL) setPrimedEquipment:(NSString *)eqKey showMessage:(BOOL)showMsg
 {
 	NSUInteger c = [eqScripts count];
 	NSUInteger current = primedEquipment;
 	primedEquipment = [self eqScriptIndexForKey:eqKey];	// if key not found primedEquipment is set to primed-none
 	BOOL result = YES;
-	if (primedEquipment == c)
+	if (primedEquipment == c && ![eqKey isEqualToString:@""])
```

Comment:
Suggest declaring a BOOL variable like unprimeRequested or any name you want, e.g. `BOOL unprimeRequested = [eqKey isEqualToString:@""];`, since you use the comparison to null string twice in the method.